.github/workflows/deploy-gh-pages.yaml

```diff
@@ -0,0 +1,54 @@
+name: "deploy-github-pages"
+
+on:
+  push:
+    branches: ["main"]
+
+permissions:
+  # So the workflow can cancel in-progress jobs
+  actions: "write"
```

Comment:
This is my mistake in previous workflow PRs,  but I found that this permission is not necessary for cancelling in-progress jobs. I think we should set these permissions to {} instead.